pkg/splunk/enterprise/indexercluster.go

```diff
@@ -322,12 +329,23 @@ func getClusterMasterStatefulSet(cr *enterprisev1.IndexerCluster) (*appsv1.State
 }
 
 // validateIndexerClusterSpec checks validity and makes default updates to a IndexerClusterSpec, and returns error if something is wrong.
+
 func validateIndexerClusterSpec(cr *enterprisev1.IndexerCluster) error {
 	// IndexerCluster must support 0 replicas for multisite / multipart clusters
 	// Multisite / multipart clusters: can't reference a cluster master located in another namespace because of Service and Secret limitations
 	if len(cr.Spec.IndexerClusterRef.Namespace) > 0 && cr.Spec.IndexerClusterRef.Namespace != cr.GetNamespace() {
 		return fmt.Errorf("Multisite cluster does not support cluster master to be located in a different namespace")
 	}
+
+	if cr.Spec.IndexerClusterRef.Name == "" {
+		err := ValidateSplunkSmartstoreSpec(&cr.Spec.SmartStore)
+		if err != nil {
+			return err
+		}
+	} else if isSmartstoreConfigured(&cr.Spec.SmartStore) {
+		return (fmt.Errorf("Smartstore configuration is not supporeted on this site. Smartstore can be configured only on site with CM"))
```

Comment:
**Comment 1.** 
Nit: s/supporeted/supported
**Comment 2.**  
I don't think so we should mention the word "site" here as it might confuse people with multi-site feature. Maybe use node or pod or resource?